mmpose/models/necks/posewarper_neck.py

```diff
@@ -0,0 +1,311 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import mmcv
+import torch
+import torch.nn as nn
+from mmcv.cnn import (build_conv_layer, build_norm_layer, constant_init,
+                      normal_init)
+from mmcv.ops import DeformConv2d
+from torch.nn.modules.batchnorm import _BatchNorm
+
+from mmpose.models.utils.ops import resize
+from ..backbones.resnet import BasicBlock, Bottleneck
+from ..builder import NECKS
+
+
+@NECKS.register_module()
+class PoseWarperNeck(nn.Module):
+    """PoseWarper neck. Paper ref: Bertasius, Gedas, et al. "Learning temporal
+    pose estimation from sparsely-labeled videos." arXiv preprint
+    arXiv:1906.04016 (2019).
+
+    <`https://arxiv.org/abs/1906.04016`>
+
+    Args:
+        in_channels (int): Number of input channels from backbone
+        out_channels (int): Number of output channels
+        inner_channels (int): Number of intermediate channels
+        deform_groups (int): Number of groups in the deformable conv
+        dilations (list|tuple): different dilations of the offset conv layers
+        extra (dict): config of the conv layer to get heatmap
+        res_blocks (dict): config of residual blocks
+        offsets (dict): config of offset conv layer
+        deform_conv (dict): config of defomrable conv layer
+        in_index (int|Sequence[int]): Input feature index. Default: 0
+        input_transform (str|None): Transformation type of input features.
+            Options: 'resize_concat', 'multiple_select', None.
+            'resize_concat': Multiple feature maps will be resize to the
+                same size as first one and than concat together.
+                Usually used in FCN head of HRNet.
+            'multiple_select': Multiple feature maps will be bundle into
+                a list and passed into decode head.
+            None: Only one select feature map is allowed.
+            Default: None.
+        freeze_trans_layer (bool): Whether to freeze the transition layer
+            (stop grad and set eval mode). Default: False.
+        norm_eval (bool): Whether to set norm layers to eval mode, namely,
+            freeze running stats (mean and var). Note: Effect on Batch Norm
+            and its variants only. Default: False
+        im2col_step: the argument `im2col_step` in deform_conv
+        least_mmcv_version (str): should use version newer than this version
+    """
+    blocks_dict = {'BASIC': BasicBlock, 'BOTTLENECK': Bottleneck}
+
+    def __init__(self,
+                 in_channels,
+                 out_channels,
+                 inner_channels,
+                 deform_groups=17,
+                 dilations=(3, 6, 12, 18, 24),
+                 extra=None,
+                 res_blocks={},
```

Comment:
The arguments `res_blocks`, `offsets` and `deform_conv` may need better design:
1. We should avoid using mutable datatype like dict as the default argument value. 
2. The docstring provides no information about what should be contained in these arguments. This could be confusing to users.
3. Use more specific names like `res_blocks_cfg`.

Reply:
OK, I will use the name `res_blocks_cfg` instead. Since the `offsets` and `deform_conv` only have one argument `kernel` in the dict,  to be more specific, I use the name `offsets_kernel` and `deform_conv_kernel` instead, and change their type to `int`.